src/solver/mlir_common.cpp

```diff
@@ -68,11 +86,11 @@ std::string ConstructBuildOptions(const ConvolutionContext& ctx,
         std::string(" --arch ") + ctx.GetStream().GetDeviceName() +
         std::string(" --groupsize ") + std::to_string(CI::GetGroupCountG(ctx)) +
         std::string(" --fil_layout ") + fil_layout +
-        std::string(" --fil_type ") + data_type +
+        std::string(" --fil_type ") + DTypeName(ctx.weights_data_type) +
```

Comment:
[future] This is cumbersome. Use ostringstream, them return `.str()`.

Reply:
Agreed and fixed